networkx/algorithms/shortest_paths/weighted.py

```diff
@@ -1935,3 +1939,200 @@ def dist_path(v):
+
+    The algorithm needs an initial solution and at every iteration it selects a
+    neigbor solution.
+
+    Considering c(x) the cost of current solution and c(x') the cost of
+    neighbor solution, then:
+    If c(x') - c(x) <= 0, the neighbor solution becomes current solution for the
+    next iteration. Otherwise, algorithm accepts the neighbor solution to become
+    current solution for the next iteration with probability
+    p = exp - ([c(x') - c(x)] / temp).
+
+    Temp is a parameter of the algorithm that represents temperature in every
+    iteration.
+
+    For more information about Simulated Annealing, see here:
+    https://en.wikipedia.org/wiki/Simulated_annealing
+
+    """
+    if not nx.is_weighted(G):
+        raise nx.NetworkXError('Given graph is not weighted.')
+
+    if not _graph_is_full_completed(G):
+        raise nx.NetworkXError('The graph you have given is not completed.')
+
+    if init_sol is not None:
+        if not _check_feasible_init_sol(G, init_sol, tmax):
+            raise nx.NetworkXError('The initial solution is not feasible.')
+
+    if temp <= 0:
+        raise nx.NetworkXError('The temperature must be greater than 0.')
+
+    best_sol = list(init_sol)
+    current_sol = list(init_sol)
+    for i in range(outer_loop):
+        for j in range(inner_loop):
+            neighbor_sol = list(_moving(G, current_sol, tmax))
+            score_of_current = _score_calculation(scores, current_sol)
+            score_of_neighbor = _score_calculation(scores, neighbor_sol)
+            df = score_of_neighbor - score_of_current
+            if df >= 0:
+                current_sol = list(neighbor_sol)
+                if  _score_calculation(scores, current_sol) > _score_calculation(scores, best_sol):
+                    best_sol = list(current_sol)
+            else:
+                p = float("{0:.2f}".format(exp(df/temp)))
+                r = float("{0:.2f}".format(random()))
+                current_sol = list(neighbor_sol) if r < p else list(current_sol)
+        temp = temp - temp * a
+    best_score = _score_calculation(scores, current_sol)
+    return best_sol, best_score, _cost_calculation(G, best_sol)
+
+def _moving(G, current_sol, tmax):
+    """
+    Method to apply a movement to the current solution in order to generate the
+    neighbor solution.
+
+    :param G: The graph of the problem. A Networkx DiGraph.
+    :param current_sol: List of nodes containing the sequence of the current
+    solution's nodes
+    :param tmax: int, The fixed amount of time.
+    :return: list, A neighbor solution
+    """
+    cs = list(current_sol)
+    current_sol_backup = list(current_sol)
+    current_sol_item = randint(1, len(current_sol) - 2)
+    helping_list = []
+    for n in G.nodes():
+        if n not in current_sol:
+            helping_list.append(n)
+    a =  randint(0, len(helping_list) - 1)
+    item = helping_list[a]
+    cs[current_sol_item], helping_list[a] = helping_list[a], cs[current_sol_item]
```

Comment:
This could just be `cs[current_sol_item] = random.choice(helping_list)`. It seems `helping_list` is not used beyond this line, so you don't need to assign to it.